### PR TITLE
Fix Chinese script code not populated for web

### PR DESCRIPTION
**Pull request by thkim1011**
Currently, the Chinese script code ('Hant' for TW and HK and 'Hans' for everything else) doesn't get populated when running web while it does for ios.

Fixes https://github.com/flutter/flutter/issues/129477.

### CODE REVIEW

**Review by ditman — APPROVED** [review]
LGTM!

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart [hunk 1]
```diff
@@ -902,19 +903,40 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
...
+    final List<String>? languages = debugLanguages ?? domWindow.navigator.languages;
     if (languages == null || languages.isEmpty) {
       // To make it easier for the app code, let's not leave the locales list
       // empty. This way there's fewer corner cases for apps to handle.
       return const <ui.Locale>[_defaultLocale];
     }
 
     final List<ui.Locale> locales = <ui.Locale>[];
+
+    // TODO(dit): Use Intl.Locale to parse each language string, instead of doing it by hand.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/platform_dispatcher.dart:924]: Can you please file an issue for this?

**Review by yjbanov — APPROVED** [review]

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart (hunk 1 above)
Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:924]: Yes!

Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:924]: https://github.com/flutter/flutter/issues/130174 Let me link it in the PR while the tree is red

**Comment by ditman** [comment]
Giving this one last go with the code sample provided by the user...

**Comment by ditman** [comment]
I got carried away tweaking this, and now it's substantially different. PTAL @yjbanov :)

**Pull request closed without merging**